bionic/provider.py

```diff
@@ -31,16 +31,23 @@ def __init__(
             self, names,
             protocols=None, code_version=None, orig_flow_name=None,
             should_persist=None, should_memoize=None, is_default_value=None,
-            docs=None):
+            docs=None, is_volatile=None):
         self.names = names
         self.protocols = protocols
         self.docs = docs
         self.code_version = code_version
         self.orig_flow_name = orig_flow_name
-        self.should_persist = should_persist
-        self.should_memoize = should_memoize
+        self._should_persist = should_persist
+        self._should_memoize = should_memoize
+        self.is_volatile = is_volatile
         self.is_default_value = is_default_value
 
+    def should_persist(self):
+        return self._should_persist and not self.is_volatile
+
+    def should_memoize(self):
+        return self._should_memoize or self.is_volatile
```

Comment:
What should happen if someone sets `@volatile(True)` and `@memoize(False)`? Maybe it's safer to throw an exception than to silently ignore the `memoize` part?

Reply:
Fair point. Users will have to add the `memoize` decorator to not memoize the value alongside `volatile`. Maybe I should call this out in the `volatile` documentation too. What do you think?

Reply:
Sorry, I didn't understand this comment. Users need to *not* add the memoize decorator, right?

Reply:
Oh I meant users will have to add both the decorators for us to be in invalid state (where memoize is false and volatile is true).